sdk/azidentity/azidentity.go

```diff
@@ -134,9 +134,9 @@ type pipelineOptions struct {
 func setAuthorityHost(authorityHost string) (string, error) {
 	if authorityHost == "" {
 		authorityHost = AzurePublicCloud
-		if envAuthorityHost := os.Getenv("AZURE_AUTHORITY_HOST"); envAuthorityHost != "" {
-			authorityHost = envAuthorityHost
-		}
+	}
+	if envAuthorityHost := os.Getenv("AZURE_AUTHORITY_HOST"); envAuthorityHost != "" {
+		authorityHost = envAuthorityHost
```

Comment:
Having this in the `authorityHost == ""` check was on purpose. This gives precedence to the secret passed in as a parameter in the ctor.

Reply:
If we leave the code like that then you can never override it via the env var.

Reply:
I left it like this because in many identity environments there are several environment variables that could possibly override the information that the user wants to employ for the credential. I am checking in with the identity crew to see if the other languages also have this, so far JS and Python both said yes that the params passed into the ctor take precendence over the environment vars. I'm asking Scott to get his thoughts on this. Will post back here once I get more information.

Reply:
I'm open to possibly having this for `AUTHORITY_HOST` but other credential params might not be able to have the same order of precedence between ctor params and env vars.

Reply:
On the one hand I can understand the params taking precedence in order to prevent them being inadvertently changed (if that was the original intent, a comment would have been helpful here; I saw the impl and assumed it was a bug).  Historically we've always had the env vars take precedence as a facility to let users change configuration on-the-fly.  I can see pros/cons for both approaches.  Adding @schaabs to weigh in.

Reply:
Just heard from Scott offline, he said "values passed through code should always trump environment variables". I can add a comment above this stating that it was a conscious decision to value parameters passed in through the code than environment variables. 

Although, since this is a concept widely applied throughout `azidentity` I can go further and state this in the README to make sure users know what is happening.

Reply:
We can also see if Scott has more to say on this thread, if you want to discuss further.

Reply:
Agreed clear documentation is required as this is a deviation from how it works in track 1.

It also means that we can't set a default value for `AuthorityHost` in the `DefaultXXXOptions()` methods else the env var will always be ignored (more behavior that needs to be documented).

Reply:
This has been resolved.  I added some comments to clarify the behavior.